.github/workflows/release.yaml

```diff
@@ -1,25 +1,38 @@
-# This is a template `release.yaml` file for ops charms
-# This file is managed by bootstack-charms-spec and should not be modified
-# within individual charm repos. https://launchpad.net/bootstack-charms-spec
-
 name: Release to Edge
 
 on:
   push:
-    branches: [master, main]
-
-concurrency:
-  group: release
-  cancel-in-progress: true
+    branches: [ main ]
+  release:
+    types: [ published ]
 
 jobs:
   check:
     uses: ./.github/workflows/check.yaml
+    secrets: inherit
 
   release:
     needs: check
-    uses: canonical/bootstack-actions/.github/workflows/charm-release.yaml@v2
-    secrets: inherit
-    with:
-      channel: "latest/edge"
-      upload-image: false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Initialize lxd  # This should dropped once it's implemented on charming-actions itself. https://github.com/canonical/charming-actions/issues/140
```

Comment:
how come we didn't need this when checking out within the bootstack action? Does destructive mode default to true?

Reply:
This part is copied from [bootstack-actions@v3](https://github.com/canonical/bootstack-actions/blob/v3/.github/workflows/charm-release.yaml#L62-L71), and I report an issue about it in official actions.

This is needed only for `destructive-mode: false`, since by default it's true.